src/history/sqlite_backed.rs

```diff
@@ -87,11 +93,14 @@ impl History for SqliteBackedHistory {
                     ":exit_status": entry.exit_status,
                     ":more_info": entry.more_info.as_ref().map(|e| serde_json::to_string(e).unwrap())
                 },
-                |row| row.get(0),
             )
-            .map_err(map_sqlite_err)?;
-        entry.id = Some(HistoryItemId::new(ret));
-        Ok(entry)
+            .map(|_| ())
+            .map_err(map_sqlite_err)
+    }
+
+    /// this history doesn't replace entries
+    fn replace(&mut self, h: &HistoryItem) -> Result<()> {
```

Comment:
They **should** use it, but this hasn't been implemented yet. For now I've left these parts untouched.
I feel like changing their behaviour would be a bad idea right now as it would change how Nushell behaves.

The method separation is useful for clarifying behaviour, and here we can clearly see a "feature" is missing, while it wasn't obvious at all previously.

For histories that support replacing, this will be useful. I added it because reedline requires histories to replace history items. It could have been a call to `update`, but it's how reedline is designed, so I preferred to not change that.